### DISCUSSION TITLE
how to enable SSL in local vuepress

### DISCUSSION THREAD

**Discussion by darkSheep404** [vuepress/core#1453] [Q&A] [answered]
hi, dear vuepress-next team
i would like ask for help about how to enable ssl in local
I can find some related info, But I wasn't familiar with vue3 and vuepress-next and the attempt didn't work

https://stackoverflow.com/questions/62977236/vuepress-how-can-i-use-https-in-dev-server
https://stackoverflow.com/questions/45807049/how-to-run-vue-js-dev-serve-with-https

**Comment by darkSheep404** [comment]
hi @Mister-Hope 
有空帮忙看看吗

**Comment by Mister-Hope** [comment]
We do not offer support for unnormal usage.

VuePress is a statIc site generator,  Which means you should build the project and use the output files for production

**Comment by meteorlxy** [comment]
You can check [vite's docs](https://vitejs.dev/config/server-options.html#server-https). VuePress dev server is based on it.

**Reply by darkSheep404** [reply, marked as answer]
thanks a lot~
by adding this in `config.js`,it works
```js
export default {
    bundler: viteBundler({
      viteOptions: {
        plugins: [
          basicSsl()
        ],
      },
      vuePluginOptions: {},
    }),
```
我在之前错误的使用单独的`vite.config.js`
应该参照这个 使用
`viteOptions`
https://v2.vuepress.vuejs.org/reference/bundler/vite.html#options

vite enable ssl refer to this
https://stackoverflow.com/questions/69417788/vite-https-on-localhost
